### DISCUSSION TITLE
Mia broke the SFC controller bindings =)

### DISCUSSION THREAD

**Discussion by giantclambake** [BlitterStudio/amiberry#1121] [General]
//School holidays here atm, my niece Mia & friend (at my behest) started playing with the emulator...
//
//A wise man once told me, 'if you can make something child-proof, the design tends to be adult-proof as well' ;)

There was a single USB connected generic SFC controller  ~ at some point during game-play, she wanted to remap the fire button  to the right shoulder button...and somehow rendered the controller nearly useless.

When I asked her to show me what she'd done, turns out she'd gone into GUI->Input->Port 1->Remap.  When she saw the image of a controller she didn't have, she went back (Esc) and changed the type from default to gamepad, and tried to remap that. When she got the same image, she figured she'd just use what buttons she had, and skip over the rest...and apparently did this a few times without success (because this isn't the way to do this =)

Regardless of how things had reached this point, it had left amiberry in a state wherein 'SFC Controller' had completely disappeared from available devices for Port 1, seemingly replaced by 'USB gamepad' instead...which you see reflected in the attached  gamecontrollerdb_user.txt file. The simple fix was to delete this file and restart amiberry, and SFC Controller returned and things were back to normal...(plus a little tute to show Mia how to do that task correctly ;)

If there is a question here, then it would probably be is this expected behavior from amiberry, when faced with parsing a gamecontrollerdb_user.txt file such as the attached (which the user has 'mistakenly' created)?

TIA

[gamecontrollerdb_user.txt.gz](https://github.com/BlitterStudio/amiberry/files/11902680/gamecontrollerdb_user.txt.gz)

**Comment by midwan** [comment]
Remapping a controller's buttons saves the results in that user-maintained file, together with the controller's unique identifier.
If an entry is found in that file for your controller, the mapping is loaded and used every time from then onwards. If you don't want that, you can always remove that line referring to your controller from the file (you don't have to delete the whole file).

So I think yes, that's expected behavior :)

**Comment by giantclambake** [comment]
Thanks  =)
